src/test/java/org/jabref/logic/bst/BibtexPurifyTest.java

```diff
@@ -1,23 +1,34 @@
 package org.jabref.logic.bst;
 
-import org.junit.jupiter.api.Test;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.fail;
 
 public class BibtexPurifyTest {
 
-    @Test
-    public void testPurify() {
-        assertPurify("i", "i");
-        assertPurify("0I  ", "0I~ ");
-        assertPurify("Hi Hi ", "Hi Hi ");
-        assertPurify("oe", "{\\oe}");
-        assertPurify("Hi oeHi ", "Hi {\\oe   }Hi ");
-        assertPurify("Jonathan Meyer and Charles Louis Xavier Joseph de la Vallee Poussin", "Jonathan Meyer and Charles Louis Xavier Joseph de la Vall{\\'e}e Poussin");
-        assertPurify("e", "{\\'e}");
-        assertPurify("Edouard Masterly", "{\\'{E}}douard Masterly");
-        assertPurify("Ulrich Underwood and Ned Net and Paul Pot", "Ulrich {\\\"{U}}nderwood and Ned {\\~N}et and Paul {\\={P}}ot");
+    @ParameterizedTest
+    @MethodSource("provideTestStrings")
+    public void testPurify(String expected, String toBePurified) {
+        assertPurify(expected, toBePurified);
+    }
+
+    private static Stream<Arguments> provideTestStrings() {
+        return Stream.of(
+                Arguments.of("i", "i"),
+                Arguments.of("0I  ", "0I~ "),
+                Arguments.of("Hi Hi ", "Hi Hi "),
+                Arguments.of("oe", "{\\oe}"),
+                Arguments.of("Hi oeHi ", "Hi {\\oe   }Hi "),
+                Arguments.of("Jonathan Meyer and Charles Louis Xavier Joseph de la Vallee Poussin", "Jonathan Meyer and Charles Louis Xavier Joseph de la Vall{\\'e}e Poussin"),
+                Arguments.of("e", "{\\'e}"),
+                Arguments.of("Edouard Masterly", "{\\'{E}}douard Masterly"),
+                Arguments.of("Ulrich Underwood and Ned Net and Paul Pot", "Ulrich {\\\"{U}}nderwood and Ned {\\~N}et and Paul {\\={P}}ot")
+        );
     }
 
     private void assertPurify(final String string, final String string2) {
```

Comment:
Please integrate in `testPurify`. We don't see why this private (!) method is there.